# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/tests/crashtests/1975995.html
# Commit: f4d3d1b63c2c
# Full Hash: f4d3d1b63c2ccf3bbcb5bf30792a9781b8f98a09
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2025-08-08 21:23:04
# Description:
#   Bug 1975995 - Fix remote track getSettings after peer connection closed. r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D260391
# ==============================================================================

diff -r 82915b38e93e -r f4d3d1b63c2c dom/media/webrtc/tests/crashtests/1975995.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/media/webrtc/tests/crashtests/1975995.html	Fri Aug 08 12:34:03 2025 +0000
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script>
+   document.addEventListener("DOMContentLoaded", async () => {
+      const peer = new RTCPeerConnection({
+        "iceServers": [{ "urls": "stun:23.21.150.121" }]
+      })
+      await peer.createOffer(() => {}, () => {}, { "offerToReceiveVideo": true })
+      peer.close()
+      const transceivers = peer.getTransceivers()
+      const transceiver = transceivers[(2540405361 % transceivers.length)]
+      const stats = await transceiver.receiver.getStats()
+      transceiver.receiver.track.getSettings()
+     setTimeout("location.reload()", Math.random() * 500)
+    })
+  </script>
+</head>
+</html>